tests/gis_tests/geoapp/tests.py

```diff
@@ -448,6 +448,17 @@ def test_relate_lookup(self):
             self.assertEqual('Texas', Country.objects.get(mpoly__relate=(pnt2, intersects_mask)).name)
             self.assertEqual('Lawrence', City.objects.get(point__relate=(ks.poly, intersects_mask)).name)
 
+        # With a complex geometry expression
+        self.assertFalse(City.objects.exclude(point__relate=(functions.Union('point', 'point'), within_mask)))
```

Comment:
This doesn't pass on Oracle: `<QuerySet [<City: Houston>, <City: Dallas>, <City: Oklahoma City>, <City: Wellington>, <City: Pueblo>, <City: Lawrence>, <City: Chicago>, <City: Victoria>]> is not false`.